File: src/reducers/timedActions.js
import T from "../constants/ActionTypes";

const initialState = {
  searchLastTriggered: null
};

export default (state=initialState, action={}) => {

  const { payload } = action;

  switch (action.type) {
    case T.SET_SEARCH_TIME:
      return {
        ...state,
        searchLastTriggered: payload
      };
    default:
      return state;
  }
};

File: src/reducers/timedActions.rs
use crate::Actions::{server, Msg};
use seed::prelude::*;

type Time = String;

#[derive(Debug, Default)]
pub struct Mdl {
    pub searchLastTriggered: Option<Time>,
}

pub fn update(action: &Msg, state: &mut Mdl, orders: &mut impl Orders<Msg>) {
    match action {
        Msg::Server(server::Msg::setSearchTime(time)) => {
            state.searchLastTriggered = Some(time.clone());
        }
        _ => {
            // do nothing
        }
    }
}
